velox/exec/Driver.cpp

```diff
@@ -479,10 +479,18 @@ StopReason Driver::runInternal(
               }
               RuntimeStatWriterScopeGuard statsWriterGuard(op);
               if (op->isFinished()) {
+                // This branch is to update the finishTiming of op.
                 auto timer =
                     createDeltaCpuWallTimer([op](const CpuWallTiming& timing) {
                       op->stats().wlock()->finishTiming.add(timing);
                     });
+              }
+
+              if (op->isFinished()) {
+                auto timer = createDeltaCpuWallTimer(
+                    [nextOp](const CpuWallTiming& timing) {
+                      nextOp->stats().wlock()->finishTiming.add(timing);
```

Comment:
> The duration of nextOp->noMoreInput() is recorded in the parent operation's stats.

Can we desribe in the PR description how above issue is fixed in this PR? It would be better to add a test on Sort or Window in [PrintPlanWithStatsTest.cpp](https://github.com/facebookincubator/velox/blob/main/velox/exec/tests/PrintPlanWithStatsTest.cpp), and paste the metrics differences in this PR.

Reply:
It seems this issue is common and not only for Sort or Window.